### DISCUSSION TITLE
Feature request: Include win-arm64 runtime

### DISCUSSION THREAD

**Discussion by andrewdewaal** [jedisct1/libsodium#1104] [Ideas]
Hello;
   We are unable to support ARM Windows devices (such as Surface Pro X) with our features requiring libsodium. I have tried (very unsuccessfully) to compile the native DLL myself and include the runtime in our CSPROJ, but haven't been able to get the compilation working. 
   
 Would it be possible to include a runtime for win-arm64?
 If not, are there any other recommendations for ARM support of libsodium to be included in a CSPROJ?
   
   Thanks very much,
Andrew

**Comment by jedisct1** [comment]
Unless it recently changed, the community version of Visual Studio doesn't seem to support ARM targets.

We can use Zig to cross-compile it, but then I have no ways to test that it actually works :(

**Comment by IIiKoBiT** [comment]
We faced the same issue. Is there any solution?

**Comment by jedisct1** [comment]
Compile with Zig:

```sh
zig build -Drelease-fast -Dtarget=aarch64-windows

ls -l zig-out/lib
```

```text
-rwxr-xr-x  1 j  staff  252928 Jun  7 17:36 sodium.dll
-rw-r--r--  1 j  staff  146964 Jun  7 17:36 sodium.lib
```

**Reply by tomuxmon** [reply]
Any chance of including it in the official build and the [nuget](https://www.nuget.org/packages/libsodium)? This would greatly reduce the complexity of our builds and increase the ease of use of this great lib.

**Reply by jedisct1** [reply]
As I don't use Windows and don't know much about NuGet, I'm still waiting for someone to tell me if the files resulting from the command above work for them.

**Reply by tomuxmon** [reply]
I have tried both `zig build -Doptimize=ReleaseFast -Dtarget=aarch64-windows` and building `builds\msvc\vs2022\libsodium\libsodium.vcxproj`. zig build results in an invalid dll. vcxproj results in a proper dll. 

I tried it with [sodium core nuget](https://www.nuget.org/packages/Sodium.Core/1.4.0-preview.1) by replacing already invalid native arm64 windows dll.

```
Unhandled exception. System.PlatformNotSupportedException: Could not initialize platform-specific components. libsodium-core may not be supported on this platform. See https://github.com/ektrah/libsodium-core/blob/master/INSTALL.md for more information.
 ---> System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (0x8007000B)
   at Interop.Libsodium.sodium_library_version_major()
   at Sodium.SodiumCore.InitializeCore()
   --- End of inner exception stack trace ---
   at Sodium.SodiumCore.InitializeCore()
   at Sodium.PublicKeyBox.GenerateKeyPair()
```
As mentioned `vcxproj` build on windows works. I wonder if if we can build the same on github CI instead of using zig build?

**Reply by alexrp** [reply]
Which Zig version did you use for this?
